### PR TITLE
feat(chat): add global cross-workspace chat session access

### LINKED ISSUE #285106: Unified management for persisted Copilot Chat history across different workspace contexts
### Problem
Copilot Chat history is currently persisted and scoped to specific workspace contexts. While this provides continuity, it creates a **management blind spot**: chat history is only accessible or deletable when that specific workspace is active.

This leads to "orphaned" chat data. For example:
* **Session A (Single file):** Open `script.js` directly -> Chat with Copilot -> Data is saved.
* **Session B (Folder):** Open the parent folder of `script.js` -> A clean slate appears.
* **Session C (Return):** Closing the folder and reopening `script.js` alone brings back Session A's history.

From a user's perspective, this data exists in a "black box." There is currently no central UI to audit or purge chat history belonging to workspaces that are not currently open. As a result, these "ghost" chat logs accumulate over time without a clear way to manage them.

### Proposal / Questions
I am seeking clarification on whether this isolation is intentional, and proposing a more transparent management workflow:
1. **Centralized History View:** Is there a plan to provide a global view (similar to "Recent Workspaces") where users can see and delete chat histories from all workspaces?
2. **Context Merging:** Should chat history from a single file be discoverable when that same file is opened within a folder context?
3. **Storage Transparency:** Where exactly is this data stored, and is there a recommended way to perform a manual "cleanup" of legacy chat data?

### Goal
To ensure VS Code remains a "clean" environment where users have full visibility and control over locally persisted AI data, preventing the long-term accumulation of unmanageable workspace state.

### Environment
- VS Code Version: 
```
$code --version
1.107.1
994fd12f8d3a5aa16f17d42c041e5809167e845a
x64
```
- Copilot Chat Extension Version: `github.copilot-chat 0.35.2`
- OS: Debian sid/Forky

### PR DESCRIPTION
**Pull request by n24q02m**
Fixes #285106

Adds the ability to **view chat sessions from other workspaces** in a read-only mode, providing unified chat history management across all VS Code workspaces.

## Problem

Currently, chat history is scoped to individual workspaces. Users cannot see or manage sessions created in other workspace contexts, leading to "orphaned" chat data that accumulates without visibility.

### Global Session Index (`ChatSessionStore`)
- **APPLICATION-scoped storage** index that tracks all sessions across workspaces
- `flushGlobalIndex()`: automatically syncs local sessions to the global index on session changes
- `getGlobalIndex()`: retrieves entries from other workspaces (excludes current workspace)
- `readCrossWorkspaceSession()`: deserializes a session from another workspace's file storage

### Service Interface (`IChatService`)
- `getCrossWorkspaceHistoryItems()`: returns session details from other workspaces
- `readCrossWorkspaceSession()`: loads a cross-workspace session's serialized data

### Cross-Workspace Controller
- `CrossWorkspaceAgentSessionsController`: implements `IAgentSessionsController` to provide a read-only view
- Implements `ISessionOpenerParticipant` to intercept session open requests for cross-workspace sessions
- Sessions opened from other workspaces are loaded as **read-only local copies** (the original is never modified)

## Architecture

```
Global Index (APPLICATION storage)
    |
    +-- Workspace A: [session-1, session-2]
    +-- Workspace B: [session-3]
    +-- Workspace C: [session-4, session-5]

When viewing from Workspace A:
    -> Shows sessions from B and C only
    -> Opens as read-only imported sessions
```

## Files Changed

| File | Change |
|------|--------|
| `chatSessionStore.ts` | +191 lines: global index logic |
| `chatService.ts` | +15 lines: interface additions |
| `chatServiceImpl.ts` | +10 lines: implementation |
| `agentSessions.contribution.ts` | +2 lines: controller registration |
| `crossWorkspaceAgentSessionsController.ts` | NEW: controller + opener participant |
| `crossWorkspaceAgentSessionsController.test.ts` | NEW: 36 tests |
| `chatSessionStore.test.ts` | +310 lines: 44 tests |
| `localAgentSessionsController.test.ts` | +8 lines: mock updates |
| `mockChatService.ts` | +9 lines: mock implementations |

## Testing

- **ChatSessionStore unit tests**: 44/44 passing
- **CrossWorkspaceAgentSessionsController unit tests**: 36/36 passing
- **All common chat tests**: 642/642 passing (no regressions)
- Test coverage includes: global index CRUD, cross-workspace session reading, read-only enforcement, edge cases (empty index, missing storage, concurrent access)

### CODE REVIEW

File: src/vs/workbench/contrib/chat/common/model/chatSessionStore.ts [hunk 1]
```diff
@@ -394,6 +401,146 @@ export class ChatSessionStore extends Disposable {
...
+			if (Object.keys(workspaceEntry.entries).length === 0) {
+				delete globalIndex.workspaces[this.workspaceId];
+			} else {
+				globalIndex.workspaces[this.workspaceId] = workspaceEntry;
+			}
+
+			this.storageService.store(GlobalChatIndexStorageKey, JSON.stringify(globalIndex), StorageScope.APPLICATION, StorageTarget.MACHINE);
+		} catch (e) {
+			this.logService.error('ChatSessionStore: Error writing global index', e);
+		}
```
[the remark this answers is not in the document]
Comment (n24q02m) [reply src/vs/workbench/contrib/chat/common/model/chatSessionStore.ts]: Good catch! Fixed - now invalidating `this.globalIndexCache = undefined` after writing to storage in `flushGlobalIndex()`.

File: src/vs/workbench/contrib/chat/test/browser/agentSessions/localAgentSessionsController.test.ts [hunk 2]
```diff
@@ -209,6 +209,14 @@ class MockChatService implements IChatService {
 	getMetadataForSession(sessionResource: URI): Promise<IChatDetail | undefined> {
 		throw new Error('Method not implemented.');
 	}
+
+	async getCrossWorkspaceHistoryItems(): Promise<any[]> {
+		return [];
+	}
```
[the remark this answers is not in the document]
Comment (n24q02m) [reply src/vs/workbench/contrib/chat/test/browser/agentSessions/localAgentSessionsController.test.ts]: Fixed - replaced any[] with the proper ICrossWorkspaceSessionDetail[] type.

File: src/vs/workbench/contrib/chat/common/model/chatSessionStore.ts [hunk 3]
```diff
@@ -61,13 +65,16 @@ export class ChatSessionStore extends Disposable {
...
 
 		const workspace = this.workspaceContextService.getWorkspace();
-		const isEmptyWindow = !workspace.configuration && workspace.folders.length === 0;
-		const workspaceId = this.workspaceContextService.getWorkspace().id;
-		this.storageRoot = isEmptyWindow ?
+		this.isEmptyWindow = !workspace.configuration && workspace.folders.length === 0;
+		this.workspaceId = workspace.id;
+		this.workspaceName = workspace.folders.length > 0
+			? workspace.folders.map(f => f.name).join(', ')
+			: workspace.configuration?.path ? workspace.configuration.path : 'Empty Window';
```
[the remark this answers is not in the document]
Comment (n24q02m) [reply src/vs/workbench/contrib/chat/common/model/chatSessionStore.ts]: Good point! Fixed - now using basename() from resources.js to extract just the workspace file name instead of the full path.
